### PR TITLE
Update the namespace exception list at multiple places 

### PR DESCRIPTION
**Pull request by jiaqiluo**
Rancher issues 
- https://github.com/rancher/rancher/issues/43078
- https://github.com/rancher/rancher/issues/42719

## Description

This PR is to add two new namespaces, `cattle-provisioning-capi-system` and `cattle-fleet-local-system`, to the exception namespace list which is mentioned at multiple places across the documentation. 

This PR also makes some general improvements:
- fix the wrong title
- add basic instructions for requirements for running rancher in hardened cluster 

## Comments

Those two new namespaces are added in rancher v2.7.7, therefore changes in this PR are applied to both the latest and versioned-v2.7 of the docs.

### CODE REVIEW

**Review by martyav — APPROVED** [review]

File: docs/pages-for-subheaders/installation-requirements.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ See our page on [best practices](../reference-guides/best-practices/rancher-serv
 
 Rancher needs to be installed on a supported Kubernetes version. Consult the [Rancher support matrix](https://www.suse.com/suse-rancher/support-matrix/all-supported-versions) to ensure that your intended version of Kubernetes is supported.
 
+### Install Rancher on a Hardened Kubernetes cluster
+
+If you install Rancher on a hardened Kubernetes cluster, check the [Exempting Required Rancher Namespaces](../../../docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/psa-config-templates.md#exempting-required-rancher-namespaces) section for detailed requirements.
```
Comment (a-blender) [inline docs/pages-for-subheaders/installation-requirements.md:28]: Both for RKE and v2/RKE2 hardened clusters?

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/psa-config-templates.md [hunk 2]
```diff
@@ -106,6 +107,8 @@ When you run Rancher on a Kubernetes cluster that enforces a restrictive securit
 - `cattle-monitoring-system`
 - `cattle-neuvector-system`
 - `cattle-prometheus`
+- `cattle-provisioning-capi-system`
+- `cattle-resources-system`
```
Comment (a-blender) [inline docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/psa-config-templates.md:111]: Why is `cattle-resources-system` also being added to exemptions ns for PSA? Applies to all references in the PR.

File: docs/pages-for-subheaders/installation-requirements.md (hunk 1 above)
Comment (jiaqiluo) [reply docs/pages-for-subheaders/installation-requirements.md:28]: Yes.

File: docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/psa-config-templates.md (hunk 2 above)
Comment (jiaqiluo) [reply docs/how-to-guides/new-user-guides/authentication-permissions-and-global-configuration/psa-config-templates.md:111]: `cattle-resources-system` is used by the app rancher-backup-restore-operator which is used in restoring or migrating Rnacher on a hardened cluster, so we need to include it in the exception list.

**Review by a-blender — APPROVED** [review]
Questions answered, lgtm!

**Pull request merged**